### PR TITLE
Session 4 uploaded please review

### PR DESCRIPTION
**Pull request by meta-aman-gautam**

### CODE REVIEW

File: Session4/ArrOperation.java [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+ * 
+ * Dated 18/07/2019
+ */
+public class ArrOperation{
+	int maxMirror(int mirrorArray[]) {
+
+		if (mirrorArray.length == 0)
+			throw new AssertionError("Array size cannot be zero");
+
+		int maxLength = 0;
```
Comment (meta-prateek-jain) [inline Session4/ArrOperation.java]: Can combine these variables declaration into one line

File: Session4/ArrOperation.java [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+	 * @param inputarray[]
+	 * @return clumps
+	 */
+	 static int noofClumps(int[] inputarray) {
+		int clumps = 0, LastIndex;
+		boolean CheckingStatus;
+
+		if (inputarray.length == 0)
+			throw new AssertionError("Array size cannot be zero");
+		for (int FirstIndex = 0; FirstIndex < inputarray.length; FirstIndex++) {
```
Comment (meta-prateek-jain) [inline Session4/ArrOperation.java]: Coding convention

File: Session4/ArrOperation.java [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+			CheckingStatus = false;
+			for (LastIndex = 1; (LastIndex + FirstIndex) < inputarray.length; LastIndex++) {
+				if (inputarray[FirstIndex] == inputarray[LastIndex + FirstIndex]) {
+					CheckingStatus = true;
+				} else
+					break;
```
Comment (meta-prateek-jain) [inline Session4/ArrOperation.java]: Should use scope brackets `{` even for the single line to keep the code clean and easily understandable

File: Session4/ArrOperation.java [hunk 4]
```diff
@@ -0,0 +1,185 @@
...
+	
+	static int countOfX(int[] inputArray, int X) {
+		int countX = 0;
+		for (int count = 0; count < inputArray.length; count++) {
+			if (inputArray[count] == X)
+				countX++;
+		}
+		return countX;
+	}
+
```
Comment (meta-prateek-jain) [inline Session4/ArrOperation.java]: why two methods, can use one method for both the purpose of counting.

File: Session4/ArrOperation.java [hunk 5]
```diff
@@ -0,0 +1,185 @@
...
+		
+		int sum = 0, leftSum = 0, rightSum = 0;
+		for (int i = 0; i < splitArray.length; i++) {
+			sum += splitArray[i];
+		}
+		if (sum % 2 != 0){
+			ReturnValue =-1;
+			return ReturnValue;
+	    }
+		int index;
```
Comment (meta-prateek-jain) [inline Session4/ArrOperation.java]: One method should have one or two return statements max.

File: Session4/ArrOperationTest.java [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+	}
+	@Test
+	public void maxMirrorTest3() {
+		int []inputArray= {1, 2, 3, 8, 9, 3, 2, 1};
+		assertEquals(3,obj.maxMirror(inputArray));
+	}
+		
+	
+	@Test
+	public void noofClumpsTest1() {
```
Comment (meta-prateek-jain) [inline Session4/ArrOperationTest.java]: Rename the method. it should be something like `testNoOfClumps_WithExpectedOutput`

File: Session4/ArrOperation.java (hunk 4 above)
Comment (meta-aman-gautam) [reply Session4/ArrOperation.java]: changed to a single method countOf_XorY

File: Session4/ArrOperation.java (hunk 3 above)
Comment (meta-aman-gautam) [reply Session4/ArrOperation.java]: changed the scope using brackets

File: Session4/ArrOperation.java (hunk 1 above)
Comment (meta-aman-gautam) [reply Session4/ArrOperation.java]: done

File: Session4/ArrOperation.java (hunk 2 above)
Comment (meta-aman-gautam) [reply Session4/ArrOperation.java]: used camel case for variable naming

File: Session4/ArrOperation.java (hunk 5 above)
Comment (meta-aman-gautam) [reply Session4/ArrOperation.java]: returnValue is the only return statement

File: Session4/ArrOperationTest.java (hunk 6 above)
Comment (meta-aman-gautam) [reply Session4/ArrOperationTest.java]: done sir

File: Session4/ArrOperation.java [hunk 7]
```diff
@@ -0,0 +1,175 @@
...
+		}
+		return maxLength;
+	}
+	
+	/**Return the number of clumps in the input array.
+	 * 
+	 * @param inputarray[]
+	 * @return clumps
+	 */
+	 static int testNoOfClumps_WithExpectedOutput(int[] inputarray) {
```
Comment (meta-prateek-jain) [inline Session4/ArrOperation.java]: What is this doing in the main class

**Pull request merged**